mlir/docs/Remarks.md

```diff
@@ -0,0 +1,212 @@
+# Remark Infrastructure
+
+Remarks are **structured, human- and machine-readable notes** emitted by the
```

Comment:
I interpret human-readable to mean "for human consumption" (eg., not `opportunity 40232csa2` but something interpretable should be produced ), is that the intent?